src/main/java/ir/sahab/zk/client/ZkClient.java

```diff
@@ -320,7 +328,14 @@ public void addPersistentNode(String path, byte[] data, boolean createParent)
         } catch (InterruptedException ex) {
             throw ex;
         } catch (Exception ex) {
-            throw new ZkClientException("Failed to add a persistent ZK node in path: " + path, ex);
+            // There is a corner case where Curator throws a ConnectionLoss but actually we added
+            // the ZK node, so when we retry we get NodeExists exception. In here, we check whether
+            // node created or not and ignore error if exists.
+            if (exists(path) && Arrays.equals(getData(path), data)) {
+                logger.warn("Retrieve error in creating ZK path but node created anyway: path = {}", path);
+            } else {
+                throw new ZkClientException("Failed to add a persistent ZK node: path = {}" + path);
```

Comment:
throw new ZkClientException("Failed to add a persistent ZK node: path : " + path, ex);
Also use : instead of = to be consistent everywhere.